internal/hcl/attribute.go

```diff
@@ -482,6 +484,47 @@ func (e *LiteralValueCollectionExpression) Value(ctx *hcl.EvalContext) (cty.Valu
 	return val, nil
 }
 
+type LiteralValueIndexExpression struct {
+	// we embed the hclsyntax.LiteralValueExpr as the hcl.Expression interface
+	// has an unexported method that we need to implement.
+	*hclsyntax.LiteralValueExpr
+	Expression *hclsyntax.IndexExpr
+}
+
+func newLiteralValueIndexExpression(expr *hclsyntax.IndexExpr) *LiteralValueIndexExpression {
+	return &LiteralValueIndexExpression{
+		LiteralValueExpr: &hclsyntax.LiteralValueExpr{Val: cty.ListValEmpty(cty.String)},
+		Expression:       expr,
+	}
+}
+
+func (e *LiteralValueIndexExpression) Value(ctx *hcl.EvalContext) (cty.Value, hcl.Diagnostics) {
+	val, diag := e.Expression.Value(ctx)
+	for _, d := range diag {
+		// if the diagnostic is an invalid index, we should try and get the first element
+		// of the collection since it should at least have the same expected type
+		if d.Summary == "Invalid index" {
+			col, colDiag := e.Expression.Collection.Value(ctx)
+
+			if !colDiag.HasErrors() && col.CanIterateElements() {
+				it := col.ElementIterator()
+				if it.Next() {
+					_, v := it.Element()
+					return v, nil
+				}
+			}
+		}
+	}
+
+	// For other diagnostics we just return an empty string
+	// as we can't determine the correct value.
+	if diag.HasErrors() {
+		return cty.StringVal(""), nil
```

Comment:
can we instead return the `mockedVal` that is passed to `mockExpressionCalls` - this keeps it consistent with the rest of the logic. It also allows us to override what mock we return (e.g. objectvalue, list, e.t.c).

The `newLiteralValueIndexExpression` can take `mockedVal` and set it as a property which we return here